src/lib/sql/sql_pipeline.cpp

```diff
@@ -53,25 +55,19 @@ SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionCont
       }
     }
 
-    try {
-      auto parsed_statement = std::make_shared<hsql::SQLParserResult>(statement);
-
-      // This statements is now owned by the parse result, which will free the memory.
-      num_statements_released++;
+    const auto statement_string_length = statement->stringLength;
+    auto parsed_statement = std::make_shared<hsql::SQLParserResult>(statement.release());
+    parsed_statement->setIsValid(true);
 
-      parsed_statement->setIsValid(true);
+    // Get the statement string from the original query string, so we can pass it to the SQLPipelineStatement
+    DebugAssert(sql_string_offset < sql.length(),
```

Comment:
How could this happen? The value returned by `statement->stringLength` would have to be wrong?